### PR TITLE
Adds sensory.dm restoration and destruction viruses

### PR DESCRIPTION
**Pull request by LanCartwright**
image

:cl:
:rscadd:Adds Sensory restoration and destruction. Effects are it's titular namesake. Both are level 6 symptoms.
/:cl:

### CODE REVIEW

**Comment by Buggy123** [comment]
:+1:

**Comment by PKPenguin321** [comment]
should probably eliminate ocular restoration/blindness symptoms and merge them with this

**Comment by kevinz000** [comment]
seems overpowered.

**Comment by PKPenguin321** [comment]
Eh, ear/brain damage and hallucinations aren't exactly that common. Hardly overpowered.

The one that makes you take damage to almost all senses, on the other hand, might be a bit much.

**Comment by LanCartwright** [comment]
I'm highly considering removing deafness from this, due to the fact that deafness.dm does actually exist.

I'm unsure what to replace it with.

**Comment by KorPhaeron** [comment]
Do me a favour for all of these viruses:

Load up a local server, infect yourself, then time how long until you are incapacitated.

**Comment by GunHog** [comment]
## Test EVERYTHING!!!!

**Comment by LanCartwright** [comment]
oh right the constant 10u of mindbreaker
fucking kek i'll change this

File: code/datums/diseases/advance/symptoms/sensory.dm [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+	..()
+	if(prob(SYMPTOM_ACTIVATION_PROB))
+		var/mob/living/M = A.affected_mob
+		switch(A.stage)
+			if(1)
+				M << "<span class='warning'><b>You can't taste a thing.</b></span>"
+			if(2)
+				M << "<span class='warning'><b>You can't anything with your fingers.</b></span>"
+			if(3)
+				M.reagents.add_reagent("ethanol"=rand(7,25))
```
Comment (RemieRichards) [inline code/datums/diseases/advance/symptoms/sensory.dm]: "ethanol", rand(7,25)
not =

**Comment by GunHog** [comment]
# Testing

File: code/datums/diseases/advance/symptoms/sensory.dm [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+
+/datum/symptom/sensory_restoration/Activate(var/datum/disease/advance/A)
+	..()
+	if(prob(SYMPTOM_ACTIVATION_PROB * 3))
+		var/mob/living/M = A.affected_mob
+		switch(A.stage)
+			if(2)
+				if(M.reagents.get_reagent_amount("inacusiate")<10)
+					M.reagents.add_reagent("inacusiate"=10)
+					M << "<span class='notice'><b>Your hearing feels clearer and crisp</b></span>"
```
Comment (tkdrg) [inline code/datums/diseases/advance/symptoms/sensory.dm]: Missing the period at the end
Also the bold text in these looks weird and inconsistent imo

File: code/datums/diseases/advance/symptoms/sensory.dm [hunk 3]
```diff
@@ -0,0 +1,95 @@
+/*
+//////////////////////////////////////
+Sensory-Restoration
+	Very very very very noticable.
+	Lowers resistance tremendously.
+	Decreases stage speed tremendously.
+	Decreases transmittablity tremendously.
+	Fatal.
```
Comment (tkdrg) [inline code/datums/diseases/advance/symptoms/sensory.dm]: how is this fatal
i don't understand what you're trying to do with this

File: code/datums/diseases/advance/symptoms/sensory.dm (hunk 2 above)
Comment (GunHog) [inline code/datums/diseases/advance/symptoms/sensory.dm]: Agreed, should lose the bold text for the heal version.

File: code/datums/diseases/advance/symptoms/sensory.dm [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+				if(M.reagents.get_reagent_amount("inacusiate")<10)
+					M.reagents.add_reagent("inacusiate"=10)
+					M << "<span class='notice'><b>Your hearing feels clearer and crisp</b></span>"
+			if(3)
+				if(M.reagents.get_reagent_amount("antihol") < 10 && M.reagents.get_reagent_amount("inacusiate") < 10 )
+					M.reagents.add_reagent_list(list("antihol"=10, "inacusiate"=10))
+					M << "<span class='notice'><b>You feel sober.</b></span>"
+			if(4)
+				if(M.reagents.get_reagent_amount("antihol") < 10 && M.reagents.get_reagent_amount("inacusiate") < 10 && M.reagents.get_reagent_amount("synaptizine") < 10)
+					M.reagents.add_reagent_list(list("antihol"=10, "inacusiate"=10, "synaptizine"=5))
```
Comment (tkdrg) [inline code/datums/diseases/advance/symptoms/sensory.dm]: not quite sure if synap is OP here, a constant pump of stun-reducing drugs sounds a bit strong

File: code/datums/diseases/advance/symptoms/sensory.dm (hunk 3 above)
Comment (GunHog) [inline code/datums/diseases/advance/symptoms/sensory.dm]: "Fatal" refers to the reagent level, not the threat of the virus itself.

File: code/datums/diseases/advance/symptoms/sensory.dm [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+
+/datum/symptom/sensory_destruction/Activate(var/datum/disease/advance/A)
+	..()
+	if(prob(SYMPTOM_ACTIVATION_PROB))
+		var/mob/living/M = A.affected_mob
+		switch(A.stage)
+			if(1)
+				M << "<span class='warning'><b>You can't taste a thing.</b></span>"
+			if(2)
+				M << "<span class='warning'><b>You can't anything with your fingers.</b></span>"
```
Comment (tkdrg) [inline code/datums/diseases/advance/symptoms/sensory.dm]: i cant what

Comment (GunHog) [inline code/datums/diseases/advance/symptoms/sensory.dm]: It is a simulation of your reduced capability to form coherent thoughts.

File: code/datums/diseases/advance/symptoms/sensory.dm (hunk 4 above)
Comment (PKPenguin321) [inline code/datums/diseases/advance/symptoms/sensory.dm]: Stimulant tho?

**Comment by LanCartwright** [comment]
Fixed grammar. Still need to debate on synaptizine, and maybe if I should adjust it to use a special kind of synaptizine.

**Pull request merged**